ReportUtilityBills/measure.rb

```diff
@@ -0,0 +1,785 @@
+      return false
+    end
+    @model = model.get
+
+    # use the built-in error checking (need model)
+    if !runner.validateUserArguments(arguments(model), user_arguments)
+      return false
+    end
+
+    # assign the user inputs to variables
+    args = get_argument_values(runner, arguments(model), user_arguments)
+    args = Hash[args.collect { |k, v| [k.to_sym, v] }]
+    args[:electricity_bill_type] = 'Simple' # TODO: support Detailed
+    output_format = args[:output_format].get
+
+    sqlFile = runner.lastEnergyPlusSqlFile
+    if sqlFile.empty?
+      runner.registerError('Cannot find EnergyPlus sql file.')
+      return false
+    end
+    @sqlFile = sqlFile.get
+    if not @sqlFile.connectionOpen
+      runner.registerError('EnergyPlus simulation failed.')
+      return false
+    end
+    @model.setSqlFile(@sqlFile)
+
+    hpxml_defaults_path = @model.getBuilding.additionalProperties.getFeatureAsString('hpxml_defaults_path').get
+    building_id = @model.getBuilding.additionalProperties.getFeatureAsString('building_id').get
+    hpxml = HPXML.new(hpxml_path: hpxml_defaults_path, building_id: building_id)
+
+    # Require full annual simulation
+    if !(hpxml.header.sim_begin_month == 1 && hpxml.header.sim_begin_day == 1 && hpxml.header.sim_end_month == 12 && hpxml.header.sim_end_day == 31)
+      runner.registerError('A full annual simulation is required for calculating utility bills.')
+      return false
+    end
+
+    # Require user-specified utility rate if 'User-Specified'
+    if args[:electricity_bill_type] == 'Detailed' && args[:electricity_utility_rate_type].get == 'User-Specified' && !args[:electricity_utility_rate_user_specified].is_initialized
+      runner.registerError('Must specify a utility rate json path when choosing User-Specified utility rate type.')
+      return false
+    end
+
+    # Require not DSE
+    (hpxml.heating_systems + hpxml.heat_pumps).each do |htg_system|
+      next unless (htg_system.is_a?(HPXML::HeatingSystem) && htg_system.is_heat_pump_backup_system) || htg_system.fraction_heat_load_served > 0
+      next if htg_system.distribution_system_idref.nil?
+      next unless htg_system.distribution_system.distribution_system_type == HPXML::HVACDistributionTypeDSE
+      next if htg_system.distribution_system.annual_heating_dse.nil?
+
+      runner.registerError('DSE is not currently supported when calculating utility bills.')
+      return false
+    end
+    (hpxml.cooling_systems + hpxml.heat_pumps).each do |clg_system|
+      next unless clg_system.fraction_cool_load_served > 0
+      next if clg_system.distribution_system_idref.nil?
+      next unless clg_system.distribution_system.distribution_system_type == HPXML::HVACDistributionTypeDSE
+      next if clg_system.distribution_system.annual_cooling_dse.nil?
+
+      runner.registerError('DSE is not currently supported when calculating utility bills.')
+      return false
+    end
```

Comment:
DRY. 95% of this code is identical to code in the `energyPlusOutputRequests` section.